Lib/gftools/fix.py

```diff
@@ -605,28 +605,30 @@ def drop_mac_names(ttfont):
     return changed
 
 
-def _blank_glyph_to_gid1(ttfont):
+def _add_blank_glyph_to_gid1(ttfont):
     from nanoemoji.reorder_glyphs import reorder_glyphs
     from nanoemoji.svg import _ensure_ttfont_fully_decompiled
-
-    second_glyph = ttfont.getGlyphOrder()[1]
+    from fontTools.ttLib.tables._g_l_y_f import Glyph
+    glyph_order = ttfont.getGlyphOrder()
     glyf_table = ttfont["glyf"]
-    # Will also work for composite glyphs since these have a -1 contour count
-    if glyf_table[second_glyph].numberOfContours > 0:
-        _ensure_ttfont_fully_decompiled(ttfont)
-        glyph_order = ttfont.getGlyphOrder()
-        empty_glyph = next(
-            (g for g in glyph_order if glyf_table[g].numberOfContours == 0),
-            None
-        )
-        if empty_glyph is None:
-            raise ValueError(
-                "Font contains no empty glyphs. Please include a space or .null glyph"
-            )
-        new_order = glyph_order
-        new_order.remove(empty_glyph)
-        new_order.insert(1, empty_glyph)
-        reorder_glyphs(ttfont, new_order)
+
+    gid1 = glyph_order[1]
+    if glyf_table[gid1].numberOfContours == 0:
+        return
+    _ensure_ttfont_fully_decompiled(ttfont)
+    hmtx = ttfont["hmtx"]
+    blank_glyph = Glyph()
+    blank_name = "blank"
+    glyf_table[blank_name] = blank_glyph
+    ttfont["glyf"] = glyf_table
+    hmtx.metrics[blank_name] = (0, 0)
+    if "HVAR" in ttfont:
+         hvar = ttfont["HVAR"]
+         hvar.table.AdvWidthMap.mapping[blank_name] = 0
```

Comment:
A few things:

1) `AdvWidthMap` is optional in HVAR, it may not be there at all (i.e. set to `None`). It happens to be in your test font, but it's not guiaranteed to be present. You should do like the subsetter does, and check `if hvar.table.AdvWidthMap:` before proceeding to update it

2) HVAR can optionally also include two additional mappings, `LsbMap` and `RsbMap` for the sidebearings (most font don't, but the spec allows those too and you should handle them here).

2) by using `0` for the empty glyph map, you are assuming that the HVAR.VarStore.VarData[0].Item[0] contains all zeros, that is no variation. This happens to be with the HVARs that fontTools.varLib generates, but it's not guaranteed that the first item in the ItemVariationStore is a no-op row of zero deltas. So it'd be safer to map it to `0xFFFFFFFF` which in OT means "no variation". You can import the constant from `fontTools.ttLib.tables.otTables import NO_VARIATION_INDEX`

Reply:
Thanks!